lang/cs/Org.Apache.REEF.Common/metrics/Api/IMetricsRecordBuilder.cs

```diff
@@ -0,0 +1,101 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+namespace Org.Apache.REEF.Common.Metrics.Api
+{
+    /// <summary>
+    /// Builder used to build a metrics record. Used by <see cref="IMetricsCollector"/> to 
+    /// add a record, typically in <see cref="IMetricsSource"/>. Note that this interface will 
+    /// be used to take Mutable metrics and convert them to immutable ones (<see cref="IImmutableMetric"/> 
+    /// so that they can be consumed possibly at some later time also.
+    /// </summary>
+    public interface IMetricsRecordBuilder
+    {
+        /// <summary>
+        /// Adds metrics tag to the record
```

Comment:
MetricsTag class...Clarified in the documentation... Are you asking why do we need tags at all....They are useful for end points (like sinks or filters) to do operations. For example an ML application running LR with LBFGS on data might like to tag the record as "ML" and "LBFGS"and "LR".